Prepare thank you banner

Abban opened the pull request:
Ticket: https://phabricator.wikimedia.org/T352651

gbirke commented:
I found out why the slides are not displaying on mobile - the Desktop banner display code fails before the mini banner initialization code is run. The culprit here is the line `setStartAnimation={this.registerStartProgressbar}` in `DesktopBanner.jsx`. The `registerStartProgressbar` method does not exist yet in the wikipedia.de version of the Dektop banner, plz compare to the Desktop wp.org banner and add code accordingly (`registerStartProgressbar`, call `this.startprogressBar` when the Dektip banner has finished animating).

How i found this: Looking at the webdev console in the browser showed an error where the property registerStartProgressbar was empty. Searching for occurrences in the code led me to DektopBanner.jsx. The console is my go-to place whenever an error occurs. many times you'll find an error message there .

Abban committed:
Prepare thank you banner

Ticket: https://phabricator.wikimedia.org/T352651

gbirke commented on banners/thank_you/banner_ctrl.de.ts:
```diff
@@ -0,0 +1,59 @@
+} );
+
+app.use( TranslationPlugin, translator );
+app.use( DynamicTextPlugin, {
```
We currently don't use the dynamic text at all. And since we don't need to project the values time-based, we might create smaller banners by leaving out this plugin completely in all entry points and just providing some formatters and `page.getCampaignParameters`

gbirke commented on banners/thank_you/content/FullPageBannerText.de.vue:
```diff
@@ -0,0 +1,53 @@
+<template>
+	<ImageWithCopyright/>
+	<p><strong>Vielen Dank! Zum Glück haben wir es trotz der aktuellen Krisen wieder geschafft.</strong></p>
+	<p> Rund { numberOfDonors } großartige Menschen haben in den letzten 57 Tagen auf unsere Spendenbitte reagiert
```
This placeholder is broken. Maybe put a prop in the component to pass the number from the banner or the entry point. Or use a message placeholder and translation key

gbirke commented on banners/thank_you/content/MiniBannerText.de.vue:
```diff
@@ -0,0 +1,5 @@
+<template>
+	<div class="wmde-banner-mini-text">
+		<p><strong>345.123 Menschen</strong> haben gespendet und Wikipedia ihre Wertschätzung erwiesen. Ihnen <strong>danken wir von ganzem Herzen!</strong></p>
```
Maybe put a prop in here to pass the number from the banner or the entry point.  Or use a message placeholder and translation key

gbirke commented on banners/thank_you/components/BannerVar.de.vue:
```diff
@@ -0,0 +1,133 @@
+				/>
+				<MembershipFormButton
+					class="hollow"
+					:label="$translate( 'call-to-action-button-different-amount' )"
```
The German text "Mit einem anderen Betrag fördern" is too big for the button at the >=100px breakpoint in the Desktop banner (it fits in the mobile one). We should probably make the buttons a bit wider on desktop, reduce the font size or make the font size dynamic so it always fits the button

gbirke committed:
Add dev and build scripts for thank you banners

Make `campaign_info.toml` with fundraising banners the default again,
reverting the previous edit to the documentation.
Add `campaign_info.thank_you.toml`
Modify Webpack configuration to check for `env.campaign_info`
parameter, use `campaign_info.toml` as the default
Add two new `npm` scripts for running a special "thank you" dev
environment and building the thank you banners
Move file reading code to CampaignConfig class
Adapt README to document the thank you functionality

gbirke reviewed (commented):
Looks good overall.

I'll add a small followup PR that'll allow us to merge this and have both regular banners and thank you banners in the same repo with 2 different campaign_info files.

Abban committed:
Hide fireworks if target is not reached

Ticket: https://phabricator.wikimedia.org/T352651

gbirke reviewed (approved):
As discussed in the daily - the text will change anyway, so my remarks will be taken care of

Abban committed:
Add win and lose copy

- Add shared banner settings
- Set fireworks to turn on only if campaign succeeds
- Show appropriate copy for success and loss

Ticket: phabricator.wikimedia.org/T352651

Abban committed:
Hook up static body on page modal open

- Also fixes incorrect event naming in banner conductors

Ticket: phabricator.wikimedia.org/T352651

Abban committed:
Update thank you banner configuration

Ticket: phabricator.wikimedia.org/T352651

Abban committed:
Final styling pass

- Fix text sizes on various window sizes
- Modify copy to fit a little better
- Emoji 🌭

Ticket: phabricator.wikimedia.org/T352651

Abban committed:
Remove click event from mini banner

And replace it with click events on the buttons

Ticket: phabricator.wikimedia.org/T352651

Abban committed:
Remove use of funds content from TY banners

Ticket: phabricator.wikimedia.org/T352651

Abban committed:
Hook TY configuration to campaign config

This will allow the banners to be updated at the last
minute without having to re-upload banners.

Ticket: https://phabricator.wikimedia.org/T353645

Abban committed:
Fix snags

KaiNissen committed:
change label for number of members

KaiNissen committed:
change number of members

Abban committed:
Remove TY banner transition

Abban committed:
Rebuild package lock
